packages/elements/src/react/utils/path-inference/next.tsx

```diff
@@ -0,0 +1,61 @@
+import { useRouter } from 'next/compat/router';
+import { useParams, usePathname } from 'next/navigation';
```

Comment:
So in `packages/nextjs/src/client-boundary/hooks/usePathnameWithoutCatchAll.tsx` we `require` these two hooks since the inline comment there says that they'd error out in pages router.

I found this papertrail:

1. https://github.com/vercel/next.js/discussions/62880#discussioncomment-9730647
2. https://github.com/vercel/next.js/issues/54242
3. https://github.com/vercel/next.js/pull/55280
4. https://github.com/vercel/next.js/blob/4398e348ee724d88282881137311680ee60fb23c/packages/next/navigation-types/compat/navigation.d.ts#L13-L20

So I think we're good with importing it directly since the version that got introduced is below our minimum Next.js version

Reply:
hmm, does that mean we can update the one within the nextjs package as well ?